variables.tf

```diff
@@ -34,6 +34,18 @@ variable "log_analytics_workspace_enabled" {
   default     = true
 }
 
+variable "log_analytics_workspace_name_override" {
+  description = "Name override of the log analytics workspace. Default is '<cluster-name>-log-analytics'"
+  type        = string
+  default     = ""
+}
+
+variable "cluster_autoscaler_diagnostic_name_override" {
```

Comment:
Please break it in two variables. `cluster_autoscaler_diagnostic_enable_override` and `cluster_autoscaler_diagnostic_override_name`